scipy/stats/_distn_infrastructure.py

```diff
@@ -1496,6 +1505,66 @@ def support(self, *args, **kwargs):
         place(out_b, 1-cond, self.badvalue)
         return out_a, out_b
 
+    def nnlf(self, theta, x):
+        """Negative loglikelihood function.
+
+        Notes
+        -----
+        This is ``-sum(log pdf(x, theta), axis=0)`` where `theta` are the
+        parameters (including loc and scale).
+        """
+        loc, scale, args = self._unpack_loc_scale(theta)
```

Comment:
`_unpack_loc_scale` is implemented by `rv_continuous` and `rv_discrete`. In `rv_discrete`, `scale` is always 1. Yes, this is "inefficient", but the inefficiency is negligible in the context of what we're doing, so I decided to do the simple thing and reuse the code we already have.